tapchannelmsg/records.go

```diff
@@ -0,0 +1,1568 @@
+package tapchannelmsg
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/btcsuite/btcd/txscript"
+	"github.com/lightninglabs/taproot-assets/asset"
+	"github.com/lightninglabs/taproot-assets/fn"
+	"github.com/lightninglabs/taproot-assets/proof"
+	"github.com/lightninglabs/taproot-assets/rfqmsg"
+	lfn "github.com/lightningnetwork/lnd/fn"
+	"github.com/lightningnetwork/lnd/input"
+	"github.com/lightningnetwork/lnd/lnwallet"
+	"github.com/lightningnetwork/lnd/tlv"
+)
+
+const (
+	// MaxNumOutputs is the maximum number of asset outputs that are allowed
+	// in a single record. This mainly affects the maximum number of asset
+	// UTXOs that can reside within a single commitment. This number should
+	// in practice be very small (probably close to 1), as all outputs must
+	// be from the same asset group but from different tranches to be
+	// encoded as an individual record.
+	MaxNumOutputs = 2048
```

Comment:
godoc recommends a value close to `1`, but isntead 2048 was chosen

what's the rationale behind this value?

Reply:
nit: wrap it in a const

Reply:
Just didn't want to introduce an upfront limitation. This is just to make sure we aren't memory-DoSed when decoding p2p messages.